### ISSUE TITLE
JustReleased misbehavior on Ubuntu family 

### ISSUE THREAD

**Issue by nvma** [EngoEngine/engo#538]
Hey, guys. I've been surprised to find out that it has been a two years (wew) since I first encountered this issue, back then we've discussed it on gitter and unfortunately nothing came out of it, I've poked around for some time but was unable to resolve it on my own.
Since then I've change several workstations and Ubuntu 18 has been released, I thought I might try again, but unfortunately the issue is still there and it's really upsetting.

**To summarize the problem**: whenever keyboard button goes through Press/Release state, right after Release it repeats the Press/Release cycle. 
I've recorded a short video to demonstrate the problem, running latest adventure demo: https://webm.xyz/webm/5ae7009561f4bc000f1671f8
With this behavior it's impossible to make any sort of keyboard controls. 
Also the following scenario doesn't work at all:
LeftButton is pressed -> we move left.
RightButton is pressed while LeftButton is pressed -> we move right
RightButton is released while LeftButton is pressed -> we should back to moving left, but we stop.

When I first encountered this issue I had an Arch machine around and I was not able to reproduce this issue there. The issue persisted through 16, 17 and now 18 Ubuntu versions.

I wasn't able to find anything in the issue tracker related to this problem and it makes me wonder whether I'm the sole Ubuntu user, which seems to be unlikely. But anyways I would really appreciate any tips that might shed light on this problem.

Thanks a lot!

**Comment by Noofbiz** [comment]
Hello @nvma! I've got a fix that works for me here https://github.com/EngoEngine/engo/tree/just-released-bug if you want to check it out

**Comment by nvma** [comment]
@Noofbiz hey, sorry that couldn't react earlier. Unfortunately no impact on my end running the latest adventure demo. 
Would you please tell me what happens on your machine when you press, say, 'D' for a ten or so seconds, when the character has already reached the edge of the world, and then release it? Does animation immediately stop?
Because for me, the longer you keep the button in pressed state, the longer it will be 'pressed' again after I release it.

**Comment by Noofbiz** [comment]
It appears to be a bug with glfw itself and is present in the C version as well [here] (https://github.com/glfw/glfw/issues/747). Looks like the fix is actually in, but is in glfw 3.3, which is currently in beta. I've got an issue here #531 to upgrade to it when it's ready. That should fix this issue. Thanks for your help and reporting!

**Comment by nvma** [comment]
@Noofbiz thank you so much for sorting this through! On a `glfw` related note, every time I'm trying to launch an Engo program I get the following message in a terminal

```
 🔥 adventure git:  go run adventure.go 
# github.com/nvma/adventure/vendor/github.com/go-gl/glfw/v3.1/glfw
In file included from glfw_linux_joystick.c:2:0:
glfw/src/linux_joystick.c: In function ‘_glfwInitJoysticks’:
glfw/src/linux_joystick.c:239:46: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 9 [-Wformat-truncation=]
             snprintf(path, sizeof(path), "%s/%s", dirname, entry->d_name);
                                              ^~
In file included from /usr/include/stdio.h:862:0,
                 from /usr/include/X11/Xcursor/Xcursor.h:26,
                 from glfw/src/x11_platform.h:38,
                 from glfw/src/internal.h:89,
                 from glfw/src/linux_joystick.c:28,
                 from glfw_linux_joystick.c:2:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 12 and 267 bytes into a destination of size 20
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
It then hangs for a about a minute before actually starting up. I've tried to google it up, but without much success. Curious of its origin, might be related?

**Comment by Noofbiz** [comment]
Yeah, this happens to me to. It used to say to report it to go-gl but that's been fixed. As for hanging for a bit, mine hung the very first time I ran it after a fresh install for a couple minutes but after that it only took a few seconds to build after the first build.

**Comment by dulsi** [comment]
Any time frame on 3.3? Alternatively could a 3.2.2 release be made? Right now glfw is not that useful on Fedora Linux due to the phantom key presses.

**Comment by Noofbiz** [comment]
Fixed by #700 ^_^
